File: plugin/src/main/groovy/com/shogo82148/ribbonizer/GreenRibbonBuilder.java
package com.shogo82148.ribbonizer;

import com.android.build.gradle.api.ApplicationVariant;
import com.shogo82148.ribbonizer.filter.ColorRibbonFilter;
import com.shogo82148.ribbonizer.resource.Resource;

import java.awt.*;
import java.io.File;
import java.util.function.Consumer;

public class GreenRibbonBuilder implements FilterBuilder {

    private static Color COLOR = new Color(0, 0x72, 0, 0x99);

    @Override
    public Consumer<Resource> apply(ApplicationVariant variant, File iconFile) {
        return new ColorRibbonFilter(variant.getBuildType().getName(), COLOR);
    }
}

File: ribbonizer-plugin/src/main/kotlin/com/shogo82148/ribbonizer/GreenRibbonBuilder.kt
package com.shogo82148.ribbonizer

import com.android.build.gradle.api.ApplicationVariant
import com.shogo82148.ribbonizer.filter.ColorRibbonFilter
import com.shogo82148.ribbonizer.resource.Resource
import com.shogo82148.ribbonizer.resource.Variant
import java.awt.Color
import java.io.File
import java.util.function.Consumer

class GreenRibbonBuilder: FilterBuilder {
    companion object {
        val COLOR: Color = Color(0, 0x72, 0, 0x99)
    }
    override fun apply(variant: Variant, iconFile: File): Consumer<Resource> {
        return ColorRibbonFilter(variant.buildType, COLOR)
    }
}